clients/scheduler/src/components/time-slot-select/ui.tsx

```diff
@@ -22,10 +22,20 @@ type SelectTimeSlotType = {
 }
 
 export const TimeSlotUi = ({ timeSlots }: UiPropsType) => {
-  const { colors, setTimeSlot, loading, shadowRoot, resetTimeSlot, providers } =
+  const { colors, date, timeSlot, setTimeSlot, preloadBooking, loading, shadowRoot, resetTimeSlot, providers } =
     useAppContext()
   const [popoverOpen, setPopoverOpen] = useState(false)
 
+  useEffect(() => {
+    if (preloadBooking && preloadBooking.start !== timeSlot.start) {
+      const bookDate = new Date(preloadBooking.start)
+      // If there's pre-booking data that hasn't been set yet
+       if (bookDate.getTime() === date.getTime()) {
+        selectTimeSlot(preloadBooking)
```

Comment:
Same - this is the component that sets the relevant states to select a timeslot and pop open the "confirm" drawer.